From bbe8050c55df48d280b08496be82467756004e30 Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Fri, 27 Sep 2024 11:00:44 +0200
Subject: [PATCH] arm64: configs: activate TEO CpuIlde governor in fragment-02

Compile the CpuIdle Timer events oriented (TEO) governor
with CONFIG_CPU_IDLE_GOV_TEO within fragment-02-defconfig-addons.config.

This simplified governor is not selected by default but it
generally safe to use it and it have some workloads and latency
benefit.

This governor can be selected with kernel command line with
`cpuidle.governor=teo` or with sysfs:

echo "teo" > /sys/devices/system/cpu/cpuidle/current_governor

Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Change-Id: I77f8b398248924a76c95d9a0224ef709ff51f289
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/407316
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 arch/arm64/configs/fragment-02-defconfig-addons.config | 1 +
 1 file changed, 1 insertion(+)

--- a/arch/arm64/configs/fragment-02-defconfig-addons.config
+++ b/arch/arm64/configs/fragment-02-defconfig-addons.config
@@ -11,6 +11,7 @@ CONFIG_CORESIGHT_SINK_TPIU=y
 CONFIG_CORESIGHT_SOURCE_ETM4X=y
 CONFIG_CORESIGHT_STM=y
 CONFIG_COUNTER=m
+CONFIG_CPU_IDLE_GOV_TEO=y
 CONFIG_CRYPTO_DEV_STM32_CRC=m
 CONFIG_CRYPTO_DEV_STM32_CRYP=m
 CONFIG_CRYPTO_DEV_STM32_HASH=m
